### PR TITLE
Swig wrapper lrx-proc -m

### PR DESCRIPTION
**Pull request by Vaydheesh**
Swig Python Wrapper for `lrx-proc -m` gaurded by `--enable-python-bindings`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,72 @@
+*.pyc
```
Comment (sushain97) [inline .gitignore]: Thanks for adding this. Silly that we didn't already have it.

File: lrx_processor.cc [hunk 2]
```diff
@@ -974,7 +974,7 @@ LRXProcessor::processME(FILE *input, FILE *output)
             continue;
           }
 
-          fwprintf(stdout, L"%S^%S/", blanks[spos].c_str(), sl[spos].c_str());
+          fwprintf(output, L"%S^%S/", blanks[spos].c_str(), sl[spos].c_str());
```
Comment (sushain97) [inline lrx_processor.cc]: Nice fix.

File: python/lex_tools.i [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+public:
+  /**
+   * Imitates functionality of lrx_proc using file path
+   */
+  void lrx_proc(char arg, char *dictionary_path, char *input_path, char *output_path);
+};
+
+
+void
+LRX::lrx_proc(char arg, char *dictionary_path, char *input_path, char *output_path)
```
Comment (sushain97) [inline python/lex_tools.i]: Is there a reason this can't just be combined with the declaration above?

File: python/lex_tools.i [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+{
+  bool useMaxEnt = false;
+  FILE *in = fopen(dictionary_path, "rb");
```
Comment (sushain97) [inline python/lex_tools.i]:
```suggestion
  FILE *dictionary = fopen(dictionary_path, "rb");
```

File: python/setup.py.in [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    sources = ['lex_tools.i']
+    cc_sources = ['lrx_processor.cc']
+    rel_path = '..'
+    sources.extend(path.join(rel_path, f) for f in cc_sources)
+    return sources
+
+def get_include_dirs():
+    # get rid of '-I' in LTTOOLBOX_CFLAGS and split list from 1, to remove extra element ''
+    # Strip extra space at end
+    dirs = '@LTTOOLBOX_CFLAGS@'.split('-I')[1:]
```
Comment (sushain97) [inline python/setup.py.in]: This seems really sketchy. What does `'@LTTOOLBOX_CFLAGS@'` actually look like? There should be a simpler way to do this.

File: python/setup.py.in [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    dirs += '@LIBXML_LIBS@'.split('-I')[1:]
+    dirs = [x.strip() for x in dirs]
+    return dirs + ['..']
+
+
+lextools_module = Extension(
+    name='_lextools',
+    sources=get_sources(),
+    swig_opts = ["-c++", "-I..", "-Wall"],
```
Comment (sushain97) [inline python/setup.py.in]: Stick with one type of quotes.

File: python/setup.py.in (hunk 5 above)
Comment (TinoDidriksen) [reply python/setup.py.in]: That is sketchy. The order of CFLAGS is not defined and there might not even be a `-I`. This should probably be a conditional regex capture instead. But it'll 99% work just fine for our builds.

File: python/lex_tools.i (hunk 3 above)
Comment (Vaydheesh) [reply python/lex_tools.i]: The class declaration is required to for wrapping. By keeping the definition separate, I simply copy paste the declaration, thus minimizing the human error (got stuck at this in past). Hence I prefer to keep the code separate. But this shouldn't be an issue now. Should I move that part in the class declaration?

File: python/setup.py.in (hunk 5 above)
Comment (Vaydheesh) [reply python/setup.py.in]: > This seems really sketchy.

`.replace('-I', '').split()` is a better solution

> What does `'@LTTOOLBOX_CFLAGS@'` actually look like?

`'-I/usr/local/include/lttoolbox-3.5 -I/usr/local/lib/lttoolbox-3.5/include -I/usr/include/libxml2 '`

I would need to use these flags in core apertium as well. Should I switch to regex?

Comment (sushain97) [reply python/setup.py.in]: `.replace('-I', '').split()` is good enough for me.

File: python/lex_tools.i (hunk 3 above)
Comment (sushain97) [reply python/lex_tools.i]: I see what you're saying. However, now you've written it 3 times! The compiler should reject any human error with mismatched interfaces anyway so let's go ahead and combine as possible :)

(On another note, these files are really annoying to read without syntax highlighting. I wonder if we can just tell GH to treat them as C++ - it doesn't support SWIG unfortunately)

**Comment by sushain97** [comment]
@TinoDidriksen could you take a quick look at this and merge if it looks good to you? I'm not really sure about the autotools stuff. Thanks!

**Pull request merged**